redeem/path_planner/PathQueue.h

```diff
@@ -134,23 +133,20 @@ class PathQueue
         // subtract one because the optimizer does touch the last index
         curTime += optimizer.beforePathRemoval(queue, readIndex, (writeIndex + queue.size() - 1) % queue.size());
 
-        Path result(std::move(queue[currentReadIndex]));
-
         readIndex = (readIndex + 1) % queue.size();
         availableSlots++;
 
         if (addPathMightBeBlocking && doesQueueHaveSpace())
         {
-            lock.unlock();
             queueHasSpace.notify_all();
         }
-        else if (availableSlots == queue.size())
+
+        if (availableSlots == queue.size())
```

Comment:
This is the really interesting change - `else if` to `if`.

Before this assumed that if the queue was out of space (so addPathMightBeBlocking is true), popping a path could not completely empty the queue. If the queue was completely filled by one big path, we wouldn't call queueIsEmpty.notify_all() even though the queue was empty after that path was removed.